swaybar/ipc.c

```diff
@@ -298,6 +304,24 @@ bool ipc_get_workspaces(struct swaybar *bar) {
 					calloc(1, sizeof(struct swaybar_workspace));
 				ws->num = json_object_get_int(num);
 				ws->name = strdup(json_object_get_string(name));
+				ws->label = strdup(ws->name);
+				// ws->num will be -1 if workspace name doesn't begin with int.
+				if(ws->num != -1) {
+					size_t len_offset = numlen(ws->num);
+					if (bar->config->strip_workspace_name) {
+						free(ws->label);
+						ws->label = malloc(len_offset + 1 * sizeof(char));
+						ws->label[len_offset] = '\0';
+						strncpy(ws->label, ws->name, len_offset);
+					} else if (bar->config->strip_workspace_numbers) {
+						len_offset += ws->label[len_offset] == ':';
+						if (strlen(ws->name) > len_offset) {
+							free(ws->label);
```

Comment:
You could remove this line and use `strcpy` below, though it's less of a fun solution as the one above.